ziti/src/main/kotlin/org/openziti/impl/ZitiContextImpl.kt

```diff
@@ -313,17 +326,33 @@ internal class ZitiContextImpl(internal val id: Identity, enabled: Boolean) : Zi
 
     }
 
-    internal fun getService(name: String): Service {
+    override fun getService(addr: InetSocketAddress): Service? = getService(addr.hostName, addr.port)
+
+    override fun getService(name: String): Service? {
         checkServicesLoaded()
-        return servicesByName.get(name) ?: throw ZitiException(Errors.ServiceNotAvailable)
+        return servicesByName.get(name)
     }
 
-    internal fun getService(host: String, port: Int): Service {
+    internal fun getService(host: InetAddress, port: Int): Service {
         checkServicesLoaded()
 
-        return servicesByName.values.find {
-            it.dns?.hostname == host && it.dns?.port == port
-        } ?: throw ZitiException(Errors.ServiceNotAvailable)
+        servicesByAddr.get(host)?.let {
```

Comment:
Will this work for CIDR addresses?

Reply:
CIDR block support will be done for tunneler, I don't see a reason to put it here